src/platform/ESP32/bluedroid/BLEManagerImpl.cpp

```diff
@@ -655,13 +655,22 @@ CHIP_ERROR BLEManagerImpl::ConfigureAdvertisingData(void)
     advData[index++] = 0x02;                            // length
     advData[index++] = CHIP_ADV_DATA_TYPE_FLAGS;        // AD type : flags
     advData[index++] = CHIP_ADV_DATA_FLAGS;             // AD value
-    advData[index++] = 0x06;                            // length
+    advData[index++] = 0x0A;                            // length
     advData[index++] = CHIP_ADV_DATA_TYPE_SERVICE_DATA; // AD type: (Service Data - 16-bit UUID)
     advData[index++] = ShortUUID_CHIPoBLEService[0];    // AD value
     advData[index++] = ShortUUID_CHIPoBLEService[1];    // AD value
-    advData[index++] = CHIP_BLE_OPCODE;                 // Used to differentiate from operational CHIP Ble adv
-    advData[index++] = static_cast<uint8_t>(advDataVersionDiscriminator >> 8); // Bits[15:12] == version - Bits[11:0] Discriminator
-    advData[index++] = static_cast<uint8_t>(advDataVersionDiscriminator & 0x00FF);
+
+    chip::Ble::ChipBLEDeviceIdentificationInfo deviceIdInfo;
+    err = ConfigurationMgr().GetBLEDeviceIdentificationInfo(deviceIdInfo);
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(DeviceLayer, "GetBLEDeviceIdentificationInfo(): ", ErrorStr(err));
+        ExitNow();
+    }
+
+    VerifyOrExit(index + sizeof(deviceIdInfo) <= sizeof(advData), err = BLE_ERROR_OUTBOUND_MESSAGE_TOO_BIG);
+    memcpy(&advData[index], &deviceIdInfo, sizeof(deviceIdInfo));
+    index = static_cast<uint8_t>(index + sizeof(deviceIdInfo));
```

Comment:
For the record I have not added VendorId/ProductId on purpose since there was still a debate to include it or not.

Is the debate finally closed ? I've not seen the debate resolution in the spec. At the moment it still says:
```Inclusion of Vendor and Product ID (VID and PID) in advertisements is under debate within the TT.```

Reply:
The implementation should reflect the currently specified payload. It's important converge on a format so that all of the examples can be commissioned.

Reply:
The answers seems to be that they are in, but optional and should be specified as zero if omitted. The length is fixed.

Reply:
> For the record I have not added VendorId/ProductId on purpose since there was still a debate to include it or not.
> 
> Is the debate finally closed ? I've not seen the debate resolution in the spec. At the moment it still says:
> `Inclusion of Vendor and Product ID (VID and PID) in advertisements is under debate within the TT.`

Yes. The debate closed a while back but the spec was written before that decision was reached. There's a PR up for the updated spec [here](https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/326)

Reply:
> The answer seems to be that they are in, but optional and should be specified as zero if omitted. The advertisement length is fixed.

Agreed on `0x0A` instead of `0x06` in this case. And I would be fine if you remove the deprecated case in the Darwin BLE connection code. 

But can we add a build flag in `_GetBLEDeviceIdentificationInfo` that defaults to `false` and that you explicitly need to turn `true` for adding those `productId` and `vendorId` informations to BLE. Otherwise the default will be to include them all the time since `productId` and `vendorId`, even if you don't intend to use them for BLE advertisement, will likely be part of the product by default.

Reply:
> > The answer seems to be that they are in, but optional and should be specified as zero if omitted. The advertisement length is fixed.
> 
> Agreed on `0x0A` instead of `0x06` in this case. And I would be fine if you remove the deprecated case in the Darwin BLE connection code.
> 
> But can we add a build flag in `_GetBLEDeviceIdentificationInfo` that defaults to `false` and that you explicitly need to turn `true` for adding those `productId` and `vendorId` informations to BLE. Otherwise the default will be to include them all the time since `productId` and `vendorId`, even if you don't intend to use them for BLE advertisement, will likely be part of the product by default.

Having that be configurable seems find.

BTW, a bunch of demos are setting these already. The immediate issue is that various commissioner/commissionee pairs **are not compatible**.

This updated PR is totally fine with zero values. I can set the ESP32 values to zero if you want.